### DISCUSSION TITLE
Duplicate Shortcut handling

### DISCUSSION THREAD

**Discussion by martinsmith1968** [Hofknecht/SystemTrayMenu#449] [Ideas]
I used to use a tiny application called Snadboys TopDesk, which is very old now and SystemTrayMenu is a much more feature rich replacement. I have a couple of directories defined - my desktop and the all-users (public) desktop, which works great but some shortcuts show up twice as they're defined in both places.

I want to add an option to "Remove Duplicate Shortcuts", which would initially work by not adding into the menu any filename + ext that had already been added. The order of the defined directories then controls which directory has precedence.

I've made the changes in a branch on my fork, and I they look like they're working, and so I can submit a PR for review soon.

I've set the text up for en-GB and it works fine, what is the best method for populating the text for other languages? I could paste the english version into google translate (or similar) and generate it that way, but I wondered if you had some guidance on a better way?

Thanks !
Martin

PS: Once this is done and accepted, I have another idea for better control of the tray icon, possibly with an animated option

**Comment by Hofknecht** [comment]
@martinsmith1968 
Thanks for the feedback!
A feature 'Remove Duplicate Shortcuts' sounds great!
(We know the problem, we solved it by making the shortcuts unique). Not showing the same shortcuts twice is a practical convenience function that also doesn't require you to move your files.
I'm considering whether the comparison of the items should perhaps also include the target path. We are also not sure whether the option should be on or off by default (I have to check whether it has a significant impact on the runtime of opening a menu) and where exactly to put it in the options menu, maybe in Customize -> Appearance or somewhere in Advanced . The designer in the options menu is a bit tricky, if you want us to link the option to that point just let us know, you can also open the PR as a draft for review.
We look forward to your contribution to the SystemTrayMenu, the PR and your next ideas! Thanks in advance.

Find the way to populate translations resx files here:
#388 

If you need help, have questions, or want to talk about it, we're here to help.
We are very happy to welcome you as a contributor to the SystemTrayMenu!
